### DISCUSSION TITLE
How can I temporarily disable dependency registration?

### DISCUSSION THREAD

**Discussion by ofersadgat** [mobxjs/mobx#3958] [General] [answered]
Hi, I'm trying to debug my application, and in doing so I want to print the value of an object at a certain point in time. The problem is that I'm trying to print the value of the property during a computation which is changing the dependency list for that computation which screws up the application. After looking through the source for reportObserved, it seems like I can hack this by doing something like: 

```ts
import { _getGlobalState } from "mobx";
function runWithoutObserving(fn: () => any){
  const globalState = _getGlobalState();
  const derivation = globalState.trackingDerivation;
  const inBatch = globalState.inBatch;
  globalState.trackingDerivation = null;
  globalState.inBatch = 0;
  fn();
  globalState.trackingDerivation = derivation;
  globalState.inBatch = inBatch;
}
function printValueWithoutObserving(observableItem: any){
  runWithoutObserving(() => {
     console.log(JSON.stringify(observableItem));
  });
}
```

but I was wondering if there is a better way of doing this than the above. Thanks!

**Comment by mweststrate** [comment, marked as answer]
That is what the untracked API is for 😀
